### PR TITLE
rados tier-3: PG Autoscaler - pool tests

### PR DESCRIPTION
**Pull request by pdhiran**
1. Verification of target size, target ratio
2. Verification of mon_target_pg_per_osd
3. Verification of min PG num for a pool

Signed-off-by: Pawan Dhiran <pdhiran@redhat.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [x] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [x] Implement the test script and perform test runs
- [x] Submit PR for code review and approve
- [x] Update Polarin Test with Automation script details and update automation fields
- [x] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pdhiran** [comment]
Logs : http://magna002.ceph.redhat.com/ceph-qe-logs/pawan/cephci-run-JYGBDS/

File: ceph/rados/pool_workflows.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+1. Autoscaler tunables
+2. Snapshots
+"""
+import logging
+import time
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+
+log = logging.getLogger(__name__)
```
Comment (tintumathew10) [inline ceph/rados/pool_workflows.py:12]: let us use the custom log class ahead.

ref : https://github.com/red-hat-storage/cephci/pull/1011

File: tests/rados/pool_tests.py [hunk 2]
```diff
@@ -11,7 +11,10 @@
 
 from ceph.ceph_admin import CephAdmin
 from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.pool_workflows import PoolFunctions
 from tests.rados.monitor_configurations import MonConfigMethods
+from tests.rados.stretch_cluster import wait_for_clean_pg_sets
+from utility.utils import method_should_succeed
 
 log = logging.getLogger(__name__)
```
Comment (tintumathew10) [inline tests/rados/pool_tests.py:19]: let us use custom log class in this file as well.

**Review by tintumathew10 — CHANGES_REQUESTED** [review]

File: ceph/rados/pool_workflows.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    def __init__(self, node: CephAdmin):
+        """
+        initializes the env to run rados commands
+        Args:
+            node: CephAdmin object
+        """
+        self.rados_obj = RadosOrchestrator(node=node)
+
+    def set_target_ratio(self, pool_name, ratio):
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply ceph/rados/pool_workflows.py]: Made use a existing method ( generic to set any param)  to set the pool property, and then verifying it in the auto-scaler status. Changed the name of this method to better convey the intent of method.

File: suites/pacific/rados/tier-3_raods_test-pool_functionalities.yaml [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        verify_pg_num_min:
+          configurations:
+            pool-1:
+              pool_name: ec_pool_2
+              pool_type: erasure
+              pg_num: 32
+              k: 4
+              m: 2
+              l: 3
+              plugin: lrc
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply suites/pacific/rados/tier-3_raods_test-pool_functionalities.yaml]: removed the LRC profile ec pool.

File: tests/rados/pool_tests.py (hunk 2 above)
Comment (pdhiran) [reply tests/rados/pool_tests.py]: Changed to use the new log class

File: ceph/rados/pool_workflows.py (hunk 1 above)
Comment (pdhiran) [reply ceph/rados/pool_workflows.py]: Changed to use new log class

File: ceph/rados/pool_workflows.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (SrinivasaBharath) [reply ceph/rados/pool_workflows.py]: ok

**Review by SrinivasaBharath — DISMISSED** [review]

**Review by tintumathew10 — DISMISSED** [review]
Looks good

**Comment by pdhiran** [comment]
Fixed the conflicts.

**Review by tintumathew10 — APPROVED** [review]

**Pull request merged**